node/src/ledger/ledger_manager.rs

```diff
@@ -0,0 +1,515 @@
+use mina_p2p_messages::v2::{
+    LedgerHash, MinaBaseAccountBinableArgStableV2, MinaBaseSparseLedgerBaseStableV2,
+    MinaLedgerSyncLedgerAnswerStableV2, MinaLedgerSyncLedgerQueryStableV1,
+    MinaStateProtocolStateValueStableV2, NonZeroCurvePoint, StateHash,
+};
+use openmina_core::block::ArcBlockWithHash;
+use openmina_core::channels::{mpsc, oneshot};
+use openmina_core::snark::{Snark, SnarkJobId};
+use std::collections::{BTreeMap, BTreeSet};
+use std::sync::{Arc, Mutex};
+
+use super::ledger_service::LedgerCtx;
+use crate::account::AccountPublicKey;
+use crate::block_producer::vrf_evaluator::DelegatorTable;
+use crate::block_producer::{BlockProducerWonSlot, StagedLedgerDiffCreateOutput};
+use crate::ledger::LedgerAddress;
+use crate::p2p::channels::rpc::StagedLedgerAuxAndPendingCoinbases;
+use crate::rpc::RpcScanStateSummaryScanStateJob;
+use crate::transition_frontier::sync::{
+    ledger::staged::StagedLedgerAuxAndPendingCoinbasesValid,
+    TransitionFrontierRootSnarkedLedgerUpdates,
+};
+use crate::transition_frontier::CommitResult;
+
+/* The type enumerating different requests that can be made to the
```

Comment:
@Sventimir Just a quick comment, here you can use `///` to make this a [doc comment](https://doc.rust-lang.org/rust-by-example/meta/doc.html) on the enum.

Reply:
That's good to know, thanks. Done.